x-pack/plugins/canvas/canvas_plugin_src/renderers/advanced_filter/component/advanced_filter.tsx

```diff
@@ -28,20 +34,20 @@ export const AdvancedFilter = ({ value, onChange, commit }) => (
         />
       </EuiFlexItem>
       <EuiFlexItem grow={false}>
-        <button
-          className="canvasAdvancedFilter__button"
-          type="submit"
-          onClick={() => commit(value)}
-        >
+        <button className="canvasAdvancedFilter__button" type="submit">
           Apply
         </button>
       </EuiFlexItem>
     </EuiFlexGroup>
   </form>
 );
 
+AdvancedFilter.defaultProps = {
+  value: '',
+};
+
 AdvancedFilter.propTypes = {
-  onChange: PropTypes.func,
+  onChange: PropTypes.func.isRequired,
```

Comment:
What are your thoughts on having both propTypes and typescript prop interfaces?  When I was typescripting react files I was replacing the former with the latter.  I suppose you lose the runtime warnings, but if you have the compile time warnings, that seems better, and then no risk of them getting out of sync - single source of truth.  But maybe there are good reason to keep both I'm unaware of.

Reply:
I'm curious about that too. I had planned to remove `propTypes` completely when converting to tsx, but I'm super new to all this type stuff and maybe there's some benefit to using both that I don't know about.

Reply:
We have to remember that TypeScript types are for build time warnings, which are transpiled away in final source, and prop types are evaluated at run time.

So prop types are incredibly important when some code is not typed with TypeScript... it's your last line of defense.

Once Canvas is fully converted, prop types become redundant.  But for now, they're the bridge between TS code and React.

Reply:
OK, so we're leaving them there so that non-ts components consuming ts components can still provide runtime warnings, but once all the components are converted, we'll be removing all the propType stuff?

Reply:
@w33ble yes indeed.  We'll have to add rules to not add `.js` files, as well... at least that would be my preference.